lib/client/indexer/user.ts

```diff
@@ -0,0 +1,375 @@
+            },
+            orders: {
+              quote_price: true,
+              filled_base_amount: true
+            },
+            prices: {
+              __args: {
+                order_by: [
+                  {
+                    created_at: "desc"
+                  }
+                ],
+                limit: 1
+              },
+              base_amount: true,
+              quote_amount: true,
+              price: true,
+              created_at: true,
+              updated_slot: true,
+              prices_type: true
+            }
+          },
+          proposal_details: {
+            title: true,
+            slug: true,
+            description: true,
+            categories: true,
+            content: true
+          }
+        }
+      }
+    })
+
+    return user_performance.map<UserPerformance | null>((p) => {
+      const proposal = p.proposal
+
+      const relatedProtocol = this.protocolMap.get(
+        proposal.dao.program.program_acct
+      );
+      const failMarket = proposal.markets.find(
+        (m) => m.market_acct === proposal.fail_market_acct
+      );
+      const passMarket = proposal.markets.find(
+        (m) => m.market_acct === proposal.pass_market_acct
+      );
+      const baseVault = proposal.conditional_vault;
+      const quoteVault = proposal.conditionalVaultByQuoteVault;
+      const proposalDetails = proposal.proposal_details[0];
+      const passPrice = passMarket?.prices ?? [];
+      const failPrice = failMarket?.prices ?? [];
+      const passTwap = passMarket?.twaps ?? [];
+      const failTwap = failMarket?.twaps ?? [];
+      const passVolume =
+        passMarket?.orders?.reduce(
+          (prev, curr) =>
+            prev +
+            PriceMath.getHumanAmount(
+              new BN(curr.filled_base_amount),
+              new BN(proposal.dao.tokenByBaseAcct?.decimals ?? 6)
+            ) *
+              curr.quote_price,
+          0
+        ) ?? 0;
+      const failVolume =
+        failMarket?.orders?.reduce(
+          (prev, curr) =>
+            prev +
+            PriceMath.getHumanAmount(
+              new BN(curr.filled_base_amount),
+              new BN(proposal.dao.tokenByBaseAcct?.decimals ?? 6)
+            ) *
+              curr.quote_price,
+          0
+        ) ?? 0;
+
+      const volume = PriceMath.getHumanAmount(
+        new BN(p.volume_bought),
+        new BN(proposal.dao.tokenByQuoteAcct?.decimals ?? 6)
+      ) + PriceMath.getHumanAmount(
+        new BN(p.volume_sold),
+        new BN(proposal.dao.tokenByQuoteAcct?.decimals ?? 6)
+      )
+
+      const pnl = PriceMath.getHumanAmount(
+        new BN(p.volume_bought),
+        new BN(proposal.dao.tokenByQuoteAcct?.decimals ?? 6)
+      ) - PriceMath.getHumanAmount(
+        new BN(p.volume_sold),
+        new BN(proposal.dao.tokenByQuoteAcct?.decimals ?? 6)
+      )
+
+      if (!relatedProtocol || !passMarket || !failMarket) return null;
+
+      return {
```

Comment:
I assume some of this was copied from the proposals client? I think that's fine for now, but maybe later we can consolidate into a helper function somewhere.